libraries/api-extractor-model/src/mixins/ApiExportedMixin.ts

```diff
@@ -0,0 +1,119 @@
+import { ApiItem, IApiItemJson, IApiItemConstructor, IApiItemOptions } from '../items/ApiItem';
+import { DeserializerContext } from '../model/DeserializerContext';
+
+/**
+ * Constructor options for {@link (IApiExportedMixinOptions:interface)}.
+ * @public
+ */
+export interface IApiExportedMixinOptions extends IApiItemOptions {
+  isExported: boolean;
+}
+
+export interface IApiExportedMixinJson extends IApiItemJson {
+  isExported: boolean;
+}
+
+const _isExported: unique symbol = Symbol('ApiExportedMixin._isExported');
+
+/**
+ * The mixin base class for API items that can be exported.
+ *
+ * @remarks
+ *
+ * This is part of the {@link ApiModel} hierarchy of classes, which are serializable representations of
+ * API declarations.  The non-abstract classes (e.g. `ApiClass`, `ApiEnum`, `ApiInterface`, etc.) use
+ * TypeScript "mixin" functions (e.g. `ApiDeclaredItem`, `ApiItemContainerMixin`, etc.) to add various
+ * features that cannot be represented as a normal inheritance chain (since TypeScript does not allow a child class
+ * to extend more than one base class).  The "mixin" is a TypeScript merged declaration with three components:
+ * the function that generates a subclass, an interface that describes the members of the subclass, and
+ * a namespace containing static members of the class.
+ *
+ * @public
+ */
+// eslint-disable-next-line @typescript-eslint/naming-convention
+export interface ApiExportedMixin extends ApiItem {
+  /**
+   * Whether the declaration is exported.
+   */
+  readonly isExported: boolean;
+
+  /** @override */
+  serializeInto(jsonObject: Partial<IApiItemJson>): void;
+}
+
+/**
+ * Mixin function for {@link (ApiExportedMixin:interface)}.
+ *
+ * @param baseClass - The base class to be extended
+ * @returns A child class that extends baseClass, adding the {@link (ApiExportedMixin:interface)} functionality.
+ *
+ * @public
+ */
+export function ApiExportedMixin<TBaseClass extends IApiItemConstructor>(
+  baseClass: TBaseClass
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+): TBaseClass & (new (...args: any[]) => ApiExportedMixin) {
+  class MixedClass extends baseClass implements ApiExportedMixin {
+    public [_isExported]: boolean;
+
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    public constructor(...args: any[]) {
+      super(...args);
+
+      const options: IApiExportedMixinOptions = args[0];
+      this[_isExported] = options.isExported;
+    }
+
+    /** @override */
+    public static onDeserializeInto(
+      options: Partial<IApiExportedMixinOptions>,
+      context: DeserializerContext,
+      jsonObject: IApiExportedMixinJson
+    ): void {
+      baseClass.onDeserializeInto(options, context, jsonObject);
+
+      const declarationReference: DeclarationReference = DeclarationReference.parse(
+        jsonObject.canonicalReference
+      );
+      options.isExported = declarationReference.navigation === Navigation.Exports;
```

Comment:
This seems potentially brittle, since the canonical reference algorithm is fairly complex, and the declaration reference parser is experimental. Since we're relying on it, maybe **ApiModelGenerator.ts** can validate that `declarationReference.navigation === Navigation.Exports` when it is writing the JSON file.

Reply:
I can understand why this code is brittle if it's relying upon experimental code. But I don't quite follow the second sentence - do you mind clarifying or pointing me to where exactly this validation would occur and how it would make this less brittle?

Reply:
I was thinking of this code, since `getDeclarationReferenceForIdentifier()` is a pretty complicated analysis:

https://github.com/microsoft/rushstack/blob/0e2502ea1e4aec2d0cbf912ed5712b8f05b0a551/apps/api-extractor/src/generators/ExcerptBuilder.ts#L150-L155

But it turns out this is not used for `ApiItem.canonicalReference`, which is built directly from API Extractor's tree by `buildCanonicalReference()`.  I think that addresses my concern. 👍